### PR TITLE
PYIC-2942: update check pending face to face verification

### PR DESCRIPTION
**Pull request by shivanshuit914**
## Proposed changes

- delete existing vc and redirect user to start journey when f2f verification is pending.
- removed old error redirection

### Issue tracking

- [2942](https://govukverify.atlassian.net/browse/PYIC-2942)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-2943]: https://govukverify.atlassian.net/browse/PYIC-2943?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java [hunk 1]
```diff
@@ -196,17 +196,25 @@ public JourneyResponse handleRequest(JourneyRequest event, Context context) {
...
-                        new StringMapMessage()
-                                .with(
-                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                        "F2F cri pending verification.");
-                LOGGER.info(message);
-                return new JourneyErrorResponse(
-                        JOURNEY_ERROR_PATH,
-                        HttpStatus.SC_INTERNAL_SERVER_ERROR,
-                        ErrorResponse.PENDING_VERIFICATION_EXCEPTION);
+            if (userHasFaceToFaceRequest) {
```
Comment (rpayal) [inline lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java]: move userHasFaceToFaceRequest & faceToFaceVc retrieve  statement (line 144 and 145) here above if block.

**Review by TalatAydinGDS — DISMISSED** [review]

File: lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java [hunk 2]
```diff
@@ -196,17 +196,25 @@ public JourneyResponse handleRequest(JourneyRequest event, Context context) {
...
+                if (Objects.isNull(faceToFaceVc)) {
+                    var message =
+                            new StringMapMessage()
+                                    .with(
+                                            LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                            "F2F cri pending verification. No VC to delete.");
+                    LOGGER.info(message);
+                } else {
+                    userIdentityService.deleteVcStoreItems(userId);
```
Comment (blakeyp) [inline lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java]: I think we should delete all their VCs? So I'm not sure this branch is needed, can let it continue as before

**Comment by blakeyp** [comment]
Isn't the ticket [2942](https://govukverify.atlassian.net/browse/PYIC-2942)?

**Comment by kerrr** [comment]
@blakeyp Yes, you are correct.

File: lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java [hunk 3]
```diff
@@ -196,17 +196,25 @@ public JourneyResponse handleRequest(JourneyRequest event, Context context) {
...
+                    var message =
+                            new StringMapMessage()
+                                    .with(
+                                            LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                            "F2F cri pending verification. Existing F2F VC deleted.");
+                    LOGGER.info(message);
+                }
+
+                return JOURNEY_NEXT;
```
Comment (blakeyp) [inline lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java]: To preserve the existing log and audit message below I think all we need to do here is log that there's a pending F2F request then let it continue

Comment (shivanshuit914) [reply lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java]: Thanks I have removed it.

**Review by blakeyp — APPROVED** [review]

**Review by kerrr — APPROVED** [review]

**Pull request merged**